system/gsettings.py

```diff
@@ -0,0 +1,225 @@
+                    uid = int(ln.split()[UID])
+                    return pwd.getpwuid(uid).pw_name
+
+        def proc_environ(pid, varname=""):
+            '''Return process environment variable of process pid'''
+            proc_env = open("/proc/{0}/environ".format(pid),'r')
+            env = proc_env.readline()
+            for env_var in env.split('\x00'):
+                if varname != "":
+                    prefix = "{0}=".format(varname)
+                    if env_var.startswith(prefix):
+                        return env_var.replace(prefix, '')
+
+        def bus_address(pid):
+            '''Return the bus address for the specified process'''
+            return proc_environ(pid, ENV_DBUS)
+
+        def new_session():
+            '''Create a new dbus session'''
+            dbus_process = subprocess.Popen('dbus-launch', stdout=subprocess.PIPE)
+            for env_string in dbus_process.stdout.readlines():
+                env_var = env_string[0:env_string.index('=')]
+                env_value = env_string[env_string.index('=')+1:]
+                os.environ[env_var] = env_value
+            return os.environ['DBUS_SESSION_BUS_PID'].replace('\n','')
+
+        # Determine the session pid.
+        self.session_pid = -1
+        self._cleanup = False
+
+        for procfile in os.listdir('/proc'):
+            if procfile.isdigit():
+                if proc_name(procfile) in SESSION_MANAGERS:
+                    self.session_pid = procfile
+
+        osuser = pwd.getpwuid(os.getuid()).pw_name
+        if self.session_pid <> -1:
+            sessionuser = proc_owner(self.session_pid)
+        else:
+            sessionuser = osuser
+
+        # Setup the environment and connect to the bus.
+        if self.session_pid <> -1 and osuser == sessionuser:
+            os.environ[ENV_DBUS] = bus_address(self.session_pid)
+        else:
+            self.session_pid = new_session()
+            self._cleanup = True
+
+        self.bus = Gio.bus_get_sync(Gio.BusType.SESSION, None)
+
+    def get_value(self, schema, key):
+        ''' Read the value of the specified key '''
+        if ':' in schema:
+            path = schema.split(':')[1]
+            schema = schema.split(':')[0]
+        else:
+            path = None
+        self.settings = Gio.Settings(schema=schema, path=path)
+        return self.settings.get_value(key)
+
+    def set_value(self, schema, key, value):
+        ''' Write a value to the specified key '''
+        if ':' in schema:
+            path = schema.split(':')[1]
+            schema = schema.split(':')[0]
+        else:
+            path = None
+        self.settings = Gio.Settings(schema=schema, path=path)
+        self.settings.set_value(key, value)
+
+    def __del__(self):
+        ''' Class cleanup '''
+        if self._cleanup:
+           # Stop the dbus daemon if one was started.
+           time.sleep(1)
+           os.kill(int(self.session_pid), signal.SIGTERM)
```

Comment:
`__del__` should be avoided if possible. If the new session should always be terminated, better install an [atexit](https://docs.python.org/2/library/atexit.html) handler or something.